app/models/osbb.rb

```diff
@@ -9,6 +9,9 @@ class Osbb < ApplicationRecord
   validates :email,  length: { maximum: 255 },
                      format: { with: VALID_EMAIL_REGEX,
                                message: 'format is not valid' }
+  has_many :members, class_name: "User", dependent: :nullify
+  has_one :lead, -> { where "role = #{User::ROLES.each_index.detect { |i| User::ROLES[i] == User::LEAD }}" },
```

Comment:
always use brackets where()

try not to interpolate in DB queries
(not really a case here but still as a reminder)
https://guides.rubyonrails.org/security.html#sql-injection 

anyway what is this??????
`User::ROLES.each_index.detect { |i| User::ROLES[i] == User::LEAD }`

you dont need a string query here use a hash way
https://apidock.com/rails/ActiveRecord/QueryMethods/where
`where(role: role_that_you_need)`

Reply:
refactor